traffic_monitor/cache/cache.go

```diff
@@ -177,54 +179,83 @@ type Filter interface {
 
 const nsPerMs = 1000000
 
-type StatComputeFunc func(resultInfo ResultInfo, serverInfo tc.LegacyTrafficServer, serverProfile tc.TMProfile, combinedState tc.IsAvailable) interface{}
+type StatComputeFunc func(resultInfo ResultInfo, serverInfo tc.LegacyTrafficServer, serverProfile tc.TMProfile, combinedState tc.IsAvailable, interfaceName string) interface{}
+type AggregateComputeFunc func(resultInfo ResultInfo) interface{}
 
-// ComputedStats returns a map of cache stats which are computed by Traffic Monitor (rather than returned literally from ATS), mapped to the func to compute them.
-func ComputedStats() map[string]StatComputeFunc {
-	return map[string]StatComputeFunc{
-		"availableBandwidthInKbps": func(info ResultInfo, serverInfo tc.LegacyTrafficServer, serverProfile tc.TMProfile, combinedState tc.IsAvailable) interface{} {
+// ComputedAggregateStats returns a map of cache stats which are computed by Traffic Monitor (rather than returned literally from ATS), mapped to the func to compute them.
+// ComputedAggregateStats returns stats based on the aggregate data from all interfaces.
+func ComputedAggregateStats() map[string]AggregateComputeFunc {
+	return map[string]AggregateComputeFunc{
+		"availableBandwidthInKbps": func(info ResultInfo) interface{} {
 			return info.Vitals.MaxKbpsOut - info.Vitals.KbpsOut
 		},
-		"availableBandwidthInMbps": func(info ResultInfo, serverInfo tc.LegacyTrafficServer, serverProfile tc.TMProfile, combinedState tc.IsAvailable) interface{} {
+		"availableBandwidthInMbps": func(info ResultInfo) interface{} {
 			return (info.Vitals.MaxKbpsOut - info.Vitals.KbpsOut) / 1000
 		},
-		"bandwidth": func(info ResultInfo, serverInfo tc.LegacyTrafficServer, serverProfile tc.TMProfile, combinedState tc.IsAvailable) interface{} {
+		"bandwidth": func(info ResultInfo) interface{} {
 			return info.Vitals.KbpsOut
 		},
-		"error-string": func(info ResultInfo, serverInfo tc.LegacyTrafficServer, serverProfile tc.TMProfile, combinedState tc.IsAvailable) interface{} {
+		"kbps": func(info ResultInfo) interface{} {
+			return info.Vitals.KbpsOut
+		},
+		"gbps": func(info ResultInfo) interface{} {
+			return float64(info.Vitals.KbpsOut) / 1000000.0
+		},
+		"loadavg": func(info ResultInfo) interface{} {
+			return info.Vitals.LoadAvg
+		},
+		"maxKbps": func(info ResultInfo) interface{} {
+			return info.Vitals.MaxKbpsOut
+		},
+	}
+}
+
+// ComputedStats returns a map of cache stats which are computed by Traffic Monitor (rather than returned literally from ATS), mapped to the func to compute them.
+func ComputedStats() map[string]StatComputeFunc {
```

Comment:
The entire return map of this function is a little verbose, IMO. If you're not using an argument, you can omit a label for it because it'll never be dereferenced. So for example you have
```go
		"availableBandwidthInMbps": func(info ResultInfo, serverInfo tc.LegacyTrafficServer, serverProfile tc.TMProfile, combinedState tc.IsAvailable, interfaceName string) interface{} {
			return (info.InterfaceVitals[interfaceName].MaxKbpsOut - info.InterfaceVitals[interfaceName].KbpsOut) / 1000
		}
```
but you're only using two of those argumentns, so this could just say:
```go

		"availableBandwidthInMbps": func(info ResultInfo, tc.LegacyTrafficServer, tc.TMProfile, tc.IsAvailable, interfaceName string) interface{} {
			return (info.InterfaceVitals[interfaceName].MaxKbpsOut - info.InterfaceVitals[interfaceName].KbpsOut) / 1000
		}
```
Which is shorter, if only a little. I wouldn't demand that you make that change - unless you want to do it for the whole thing, removing names of unused parameters to make things easier to read.